src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3492,19 +3617,84 @@ private string ScanNumberHelper(char firstChar, out bool hex, out bool real, out
             if (c.IsTypeSuffix())
             {
                 SkipChar();
-                suffix = c;
+                switch (c)
+                {
+                    case 'l':
+                    case 'L':
+                        suffix |= NumberSuffixFlags.Long;
+                        break;
+                    case 'd':
+                    case 'D':
+                        suffix |= NumberSuffixFlags.Decimal;
+                        break;
+                    case 'u':
+                    case 'U':
+                        suffix |= NumberSuffixFlags.Unsigned;
+                        break;
+                    case 's':
+                    case 'S':
+                        suffix |= NumberSuffixFlags.Short;
+                        break;
+                }
+
                 c = PeekChar();
+                if (c.IsTypeSuffix())
+                {
+                    SkipChar();
+
+                    if (suffix == NumberSuffixFlags.Unsigned)
+                    {
+                        switch (c)
+                        {
+                            case 'l':
+                            case 'L':
+                                suffix |= NumberSuffixFlags.Long;
+                                break;
+                            case 's':
+                            case 'S':
+                                suffix |= NumberSuffixFlags.Short;
+                                break;
+                            default:
+                                notNumber = true;
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        notNumber = true;
+                    }
+
+                    c = PeekChar();
+                }
             }
 
             if (c.IsMultiplierStart())
             {
                 SkipChar();
 
-                if (c == 'k' || c == 'K') { multiplier = 1024; }
-                else if (c == 'm' || c == 'M') { multiplier = 1024 * 1024; }
-                else if (c == 'g' || c == 'G') { multiplier = 1024 * 1024 * 1024; }
-                else if (c == 't' || c == 'T') { multiplier = 1024L * 1024 * 1024 * 1024; }
-                else if (c == 'p' || c == 'P') { multiplier = 1024L * 1024 * 1024 * 1024 * 1024; }
+                switch (c)
```

Comment:
Going from 5 lines to 22 lines of code is an improvement? 🙂 
Theoretically a switch might be faster if the jit generates a jump table, but if performance were the concern, I'd use a statically initialized array and replace these 5 lines of code with:
```
multiplier = multipliers[c];
```

Reply:
Ooh, that's a good idea! I'll borrow that and submit it in the next PR. Feel free to review this further if you want, but be aware that 90% of the code here has been completely overhauled in the upcoming PR (and it has fancy tidbits with binary parsing and underscore support yay!) for y'all to have fun combing through as I re-learn my C# ;)

Reply:
Roger that! :)